builds/misc/templates/consolidate-edgehub-artifacts.yaml

```diff
@@ -0,0 +1,24 @@
+# This template will move the mqttd and watchdog artifacts into the EdgeHub dotnet publish folder
+# EdgeHub's dotnet publish folder will serve as a minimal build context for the image
+parameters:
+  dotnet.artifacts.path: 'publish/Microsoft.Azure.Devices.Edge.Hub.Service'
+  mqttd.artifacts.path: 'mqtt/target/'
+  mqttd.config.path: 'mqtt/mqtt-broker/config/production.json'
+  watchdog.artifacts.path: 'edge-hub/watchdog/target'
+  all.artifacts.path: 'publish/edge-hub'
+
+steps:
+  - bash: |
+      mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}
+      mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/mqttd
+      mkdir -p $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/watchdog
+
+      cp  -r ${{ parameters['watchdog.artifacts.path'] }} $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/watchdog
```

Comment:
nit: not sure if it is possible or if it would work, but for readability you can try to define `variables` block after params and define vars such as `watchdog.dest.path` and similar.
```
variables:
  watchdog.dest.path: $(Build.BinariesDirectory)/${{ parameters['all.artifacts.path'] }}/watchdog
 ***
```

Reply:
I tried adding a variables section before and got an error. I think it was because you can't use variables in a template?